File: packages/backend/src/main/kotlin/hollybike/api/types/association/TAssociation.kt
/*
  Hollybike API Kotlin KTor Graalvm application
  Made by MacaronFR (Denis TURBIEZ) and Loïc Vanden Bossche
*/
package hollybike.api.types.association

import hollybike.api.repository.Association
import kotlinx.serialization.Serializable

@Serializable
data class TAssociation(
	val id: Int,
	val name: String,
	val status: EAssociationsStatus,
	val picture: String? = null
) {
	constructor(entity: Association) : this(
		entity.id.value,
		entity.name,
		entity.status,
		entity.signedPicture
	)
}

File: packages/frontend/src/types/TAssociation.ts
/*
  Hollybike Back-office web application
  Made by MacaronFR (Denis TURBIEZ) and enzoSoa (Enzo SOARES)
*/
import { EAssociationStatus } from "./EAssociationStatus.ts";

export interface TAssociation {
	id: number,
	name: string,
	status: EAssociationStatus,
	picture?: string
}

export const dummyAssociation: TAssociation = {
	id: -1,
	name: "dummyAssociation",
	status: EAssociationStatus.Enabled,
};
